fvm/transactionSequenceNum_test.go

```diff
@@ -0,0 +1,112 @@
+package fvm_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/onflow/flow-go-sdk/crypto"
+	"github.com/onflow/flow-go/fvm"
+	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/fvm/programs"
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/fvm/utils"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTransactionSequenceNumProcess(t *testing.T) {
+	t.Run("sequence number update (happy path)", func(t *testing.T) {
+		ledger := utils.NewSimpleView()
+		sth := state.NewStateHolder(state.NewState(ledger))
+		accounts := state.NewAccounts(sth)
+
+		// create an account
+		address := flow.HexToAddress("1234")
+		privKey, err := unittest.AccountKeyDefaultFixture()
+		require.NoError(t, err)
+		err = accounts.Create([]flow.AccountPublicKey{privKey.PublicKey(1000)}, address)
+
+		tx := flow.TransactionBody{}
+		tx.SetProposalKey(address, 0, 0)
+		signEnvelope(&tx, address, *privKey)
```

Comment:
signatures are not needed for this test but I added them here, so in the future when we move proposal signature verification to this component, the test will check it.